main/lsp/requests/completion.cc

```diff
@@ -254,27 +262,39 @@ string methodSnippet(const core::GlobalState &gs, core::SymbolRef method, core::
 unique_ptr<CompletionItem> getCompletionItemForKeyword(const core::GlobalState &gs, const LSPConfiguration &config,
                                                        const RubyKeyword &rubyKeyword, const core::Loc queryLoc,
                                                        string_view prefix, size_t sortIdx) {
-    auto label = rubyKeyword.keyword;
-    auto item = make_unique<CompletionItem>(label);
+    auto item = make_unique<CompletionItem>(rubyKeyword.keyword);
     item->sortText = fmt::format("{:06d}", sortIdx);
-    item->kind = CompletionItemKind::Keyword;
 
     // TODO(jez) This should probably be a helper function (see getCompletionItemForSymbol)
     u4 queryStart = queryLoc.beginPos();
     u4 prefixSize = prefix.size();
+
+    string replacementText;
+    if (rubyKeyword.snippet.has_value() && config.getClientConfig().clientCompletionItemSnippetSupport) {
+        item->insertTextFormat = InsertTextFormat::Snippet;
+        item->kind = CompletionItemKind::Snippet;
+        replacementText = rubyKeyword.snippet.value();
+    } else {
+        item->insertTextFormat = InsertTextFormat::PlainText;
+        item->kind = CompletionItemKind::Keyword;
+        replacementText = rubyKeyword.keyword;
+    }
+
     auto replacementLoc = core::Loc{queryLoc.file(), queryStart - prefixSize, queryStart};
     auto replacementRange = Range::fromLoc(gs, replacementLoc);
-    auto replacementText = label;
     if (replacementRange != nullptr) {
         item->textEdit = make_unique<TextEdit>(std::move(replacementRange), replacementText);
     } else {
-        // TODO(jez) Why is replacementRange nullptr? instrument this and investigate when it fails
+        // Range::fromLoc failed... maybe the fuzzer is running?
         item->insertText = replacementText;
     }
-    item->insertTextFormat = InsertTextFormat::PlainText;
 
-    item->documentation =
-        make_unique<MarkupContent>(config.getClientConfig().clientCompletionItemMarkupKind, rubyKeyword.documentation);
+    item->detail = fmt::format("(sorbet) {}", rubyKeyword.documentation);
+    if (rubyKeyword.snippet.has_value()) {
```

Comment:
Should this be guarded on `config.getClientConfig().clientCompletionItemSnippetSupport` as well?